### PR TITLE
Add Previous Home Owners to Re-Registration flow

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#21247

*Description of changes:*
- Create a new common CollapsibleCard component 
- Add Previous Owners to Home Owners step in Re-Registration flow
- Unit tests

Screenshot 2024-06-13 at 03 12 36

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1934-3bdqy1kh.web.app

File: ppr-ui/src/components/common/CollapsibleCard.vue
Comment (dimak1) [inline ppr-ui/src/components/common/CollapsibleCard.vue:1]: There might be a better name for this component, if you have one, lmk! @cameron-eyds

Comment (cameron-eyds) [reply ppr-ui/src/components/common/CollapsibleCard.vue:1]: Seems good to me!

File: ppr-ui/src/store/state/state-model.ts [hunk 1]
```diff
@@ -253,6 +253,9 @@ export const stateModel: StateModelIF = {
       otherRemarks: ''
     }
   },
+
+  // mhrReRegistrationPreviousOwnerGroups?: null,
+  // mhrReRegistrationPreviousOwnerGroups?: null,
```
Comment (cameron-eyds) [inline ppr-ui/src/store/state/state-model.ts]: Some dev code here

**Review by cameron-eyds — APPROVED** [review]
Looks good!

File: ppr-ui/src/store/state/state-model.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/store/state/state-model.ts]: Removed

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1934-3bdqy1kh.web.app

**Review by doug-lovett — APPROVED** [review]
Looks great.

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue [hunk 2]
```diff
@@ -780,7 +782,9 @@ export default defineComponent({
     isMhrTransfer: { type: Boolean, default: false },
     hideRemovedOwners: { type: Boolean, default: false },
     showChips: { type: Boolean, default: false },
-    validateTransfer: { type: Boolean, default: false }
+    validateTransfer: { type: Boolean, default: false },
+    hideTableErrors: { type: Boolean, default: false },
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:786]: @cameron-eyds added these additional props to:
1. Prevent errors showing up (they were showing up in Prev Owners if actual Home Owners had errors)
2. Force show groups for TIC tenancy types
I was thinking to combine these two into one prop (eg isPrevHomeOwners) but decided to keep it more flexible

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1934-3bdqy1kh.web.app

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:786]: Sounds good!

**Review by cameron-eyds — APPROVED** [review]
Still looks good to me!

**Pull request merged**